include/sensors/core/tasks/pressure_driver.hpp

```diff
@@ -241,20 +247,18 @@ class MMR920C04 {
     }
 
     auto set_sync_bind(SensorOutputBinding binding) -> void {
-        _output_sync_bind = binding;
-        if (binding == SensorOutputBinding::none) {
-            // NOLINTNEXTLINE
-            stop_polling = true;
-        } else {
-            stop_polling = false;
-        }
         hardware.reset_sync();
+        stop_polling = ((static_cast<uint8_t>(binding) & 0x3) == 0x0);
+        report = ((static_cast<uint8_t>(binding) & 0x2) == 0x2);
```

Comment:
these similarly should be helper functions and they should compare to the actual enum values casted into uint8_ts, not literals of the underlying values.